src/client/html/script_loader.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <queue>
+#include <vector>
+#include <memory>
+#include <functional>
+
+namespace dom
+{
+  class HTMLScriptElement;
+  class Document;
+
+  /**
+   * ScriptLoader manages the loading and execution order of scripts according to HTML specification.
+   * It handles:
+   * - Normal scripts: Block parsing until loaded and executed
+   * - Async scripts: Load in parallel, execute immediately when ready
+   * - Defer scripts: Load in parallel, execute in order after parsing completes
+   */
+  class ScriptLoader
```

Comment:
Please consider to make this class to be a execution manager so that it doesn't depend on the `HTMLScriptElement`.